### ISSUE TITLE
Clickable row?

### ISSUE THREAD

**Issue by risalfajar** [bryanmylee/svelte-headless-table#104]
Is any plugin to add click event to each row with its corresponding data as argument?

**Comment by bryanmylee** [comment]
Could you add an event handler on your `<tr/>` element?

If not, you would have to write a custom plugin to do so.

**Comment by risalfajar** [comment]
Yes currently it's possible to implement this

```svelte
{#each $rows as row (row.id)}
    <Subscribe rowAttrs={row.attrs()} let:rowAttrs>
        <tr {...rowAttrs} class:clickable on:click={() => dispatch('click', row.original)}>
            {#each row.cells as cell (cell.id)}
                <Subscribe attrs={cell.attrs()} let:attrs>
                    <td {...attrs}>
                        <Render of={cell.render()}/>
                    </td>
                </Subscribe>
            {/each}
        </tr>
    </Subscribe>
{/each}
```

But maybe having a plugin to do this is a nice to have.

**Comment by bryanmylee** [comment]
Yeah it would be nice to have a plugin to attach listeners to table elements. I'll look into it!

**Comment by fstodulski** [comment]
<img width="1002" alt="Screenshot 2024-01-09 at 18 25 51" src="https://github.com/bryanmylee/svelte-headless-table/assets/14015728/e41d2ad6-296e-4901-939e-be4a606f9a09">
Hey! is it possible to add simple way to pass an ID for Navigating Purposes? 
Also, `row.original` seems to have an issue with types.

**Comment by lfsc09** [comment]
Hey @fstodulski, take a look at https://github.com/bryanmylee/svelte-headless-table/issues/31#issuecomment-1167195138.
I've resolved my type issues with `row.original` with it.
